From 05abbb3dc82d937a5c68d7ffcbea578e0dff58ed Mon Sep 17 00:00:00 2001
From: Debasish Biswas <debasishbsws.dev@gmail.com>
Date: Wed, 15 Jan 2025 00:14:35 +0530
Subject: [PATCH] Fix BUILD v4.0.0: Update Boost API usage: Replace reset() with restart()

In boost 1.87 the API changed and upstream uses the older API: replace
the deprecated ioContext.reset() with ioContext.restart().

Additionally, in more recent Boost versions, the boost::asio usage in
this file has also changed.

Signed-off-by: Debasish Biswas <debasishbsws.dev@gmail.com>
Signed-off-by: Daniel Watkins <daniel.watkins@chainguard.dev>
---

diff --git a/common/ChildProcess/IoContext.cpp b/common/ChildProcess/IoContext.cpp
index 56f1422..cb3f6b0 100644
--- a/common/ChildProcess/IoContext.cpp
+++ b/common/ChildProcess/IoContext.cpp
@@ -24,6 +24,8 @@
 #include <optional>
 #include <thread>
 
+#include <boost/asio/post.hpp>
+
 struct IoContext::Impl
 {
     boost::asio::io_context ioContext;
@@ -48,7 +50,7 @@ struct IoContext::Impl
                 while (!cancellationToken.stop_requested())
                 {
                     ioContext.run_for(boost::asio::chrono::milliseconds(500));
-                    ioContext.reset();
+                    ioContext.restart();
                 }
             }
             catch (std::exception const &e)
@@ -113,5 +115,5 @@ std::mutex &IoContext::GetProcessCreationMutex()
 
 void IoContext::Post(std::function<void()> func)
 {
-    m_impl->Get().post(func);
-}
\ No newline at end of file
+    boost::asio::post(m_impl->Get(), func);
+}
